### PR TITLE
Support async benchmarks

### LINKED ISSUE #203: Benchmarking `async` functions?
Hello again! 😁 

We're looking for something to benchmark Cats Effect on JS, and of course you have the perfect project for this :) The trickyness for our use-case however is that we need to be able to benchmark an async function, e.g. something that returns a `Future` or a `js.Promise` or asynchronously invokes a callback. I took a walk through your source, which is really excellent btw, but it seems like to accommodate this would require quite a bit of changes to the `Benchmark`, `Engine`, and `Clock` interfaces/implementations at least. Is this something you'd be open to, or is there a problem with benchmarking an asynchronous function that I'm not realizing?

Thanks in advance!

### PR DESCRIPTION
**Pull request by japgolly**
Closes #203

### CODE REVIEW

**Comment by armanbilge** [comment]
This _was_ extremely straightforward!! Speaks to the excellent underlying design ;) Thanks for the incredibly fast turnaround 🤩

File: demo/src/main/scala/demo/suites/shootouts/AsyncEffectShootout.scala [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import demo.TempExt._
+import demo.Util._
+import japgolly.scalajs.benchmark._
+import japgolly.scalajs.benchmark.gui._
+import japgolly.scalajs.react.callback.AsyncCallback
+import japgolly.scalajs.react.vdom.html_<^._
+import scala.scalajs.js
+import scala.scalajs.js.|
+
+object AsyncEffectShootout {
```
Comment (armanbilge) [inline demo/src/main/scala/demo/suites/shootouts/AsyncEffectShootout.scala:13]: Exciting!!

**Comment by japgolly** [comment]
> This was extremely straightforward!! Speaks to the excellent underlying design ;) Thanks for the incredibly fast turnaround

hehe thanks! Yeah it actually took me longer to add the Async Effect shootout lol. Speaking of which, this is how it went:

Screenshot_2021-08-16_12-32-51

Surprising results!

**Comment by armanbilge** [comment]
Wow you are right, CE3 got smoked!! Lol. Which browser is this btw?

These are my results on Firefox.
```
Benchmark     Library                     Size     Cnt      ms/op ±  error       ops/s ±   error
                                                                                                
FlatMap       Cats Effect v3.2.2           500      10     23.003 ± 10.352      47.427 ±  22.936
FlatMap       JS Promise                   500      10      4.123 ±  3.318     274.770 ± 106.401
FlatMap       ScalaJsReact v2.0.0-RC3      500      10      4.307 ±  0.419     232.980 ±  20.827
FlatMap       ZIO v1.0.10                  500      10      5.416 ±  0.505     185.248 ±  16.600
```

**Comment by japgolly** [comment]
Woah, at first I was like *"wow Firefox is FAST!"* and then I realised the units are different (ms vs us) and then I was like *"wow Firefox is SLOW!"* :laughing: 

My results were from Chrome. If you're running locally, use `bin/serve-demo` so that Chrome starts in a more time-precise mode

**Comment by japgolly** [comment]
And I'm not happy seeing ZIO beat scalajs-react. That was surprising and sad for me :( Good on them though, they must be doing something really well

**Pull request merged**